Content/Items/Vitric/Weapons.CoachGunUpgrade.cs

```diff
@@ -0,0 +1,686 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using StarlightRiver.Content.Buffs;
+using StarlightRiver.Content.Dusts;
+using StarlightRiver.Content.Items.Misc;
+using StarlightRiver.Core;
+using StarlightRiver.Core.Systems;
+using StarlightRiver.Helpers;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Terraria;
+using Terraria.Audio;
+using Terraria.DataStructures;
+using Terraria.Graphics.Effects;
+using Terraria.ID;
+using Terraria.ModLoader;
+namespace StarlightRiver.Content.Items.Vitric
+{
+    public class CoachGunUpgrade : ModItem
+    {
+        private int cooldown = 0;
+
+        public override void HoldItem(Player player) => cooldown--;
+        public override bool AltFunctionUse(Player player) => true;
+
+        public override string Texture => AssetDirectory.VitricItem + Name;
+
+        public override void Load()
+        {
+            GoreLoader.AddGoreFromTexture<SimpleModGore>(Mod, AssetDirectory.VitricItem + "CoachGunUpgradeCasing");
+        }
+        public override void SetStaticDefaults()
+        {
+            DisplayName.SetDefault("Magmatic Coach Gun"); //placeholder i couldnt think of a name, also sweltered needs a keyword cause this tooltip is already long as fuck
+            Tooltip.SetDefault("Press <right> to throw out an unstable crystal bomb\nExplodes shortly after, causing all other crystal bombs to also explode\n" +
```

Comment:
This is quite wordy. It would probably be wise to have someone re-write this

Reply:
I believe egshels is going to rewrite it